aten/src/ATen/native/mps/operations/ScatterGather.mm

```diff
@@ -83,6 +85,22 @@
         getInput = [mpsGraph sliceTensor:inputTensor starts:starts ends:ends strides:strides name:nil];
       }
 
+      // PyTorch issue #135240: MPS reshape underneath goes haywire in
+      // gatherAlongAxis. Fix is coming in future OS, until then requires workaround.
+      // If this op is failing and isMacos15_3 = 1, the OS level fix has not
+      // landed and we'll need to increment the version check until it does.
+      bool workaroundSingleDim = (self_arg.squeeze().sizes().size() == 1 && self_arg.sizes().size() > 1);
+      bool isMacos15_3 = is_macos_13_or_newer(MacOSVersion::MACOS_VER_15_3_PLUS);
```

Comment:
This feels wrong to me. We don't know what will or will not be fixed in macOS-15.3, so let's not condition this workaround. When OS is released we can add the check and run tests